utils/profdata_merge_worker.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python
+
+# utils/profdata_merge_worker.py
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+# This script is used to prevent profile data filling up available disk space
+# by listening for profile data and merging them into a universal profdata
+# file while tests are executing.
+
+import os
+import shutil
+import pipes
+import sys
+import socket
+import SocketServer
+from multiprocessing import Lock, Process, Queue, JoinableQueue
+import thread
+import subprocess
+import tempfile
+import argparse
+
+SERVER_ADDRESS = ('localhost', 12400)
+TESTS_FINISHED_SENTINEL = "PROFDATA_MERGE_WORKER_TESTS_FINISHED_SENTINEL"
+FILE_QUEUE = JoinableQueue()
+FILES_MERGED = set()
+CONFIG = None
+
+class Config():
+    """A class to store configuration information specified by command-line arguments.
+    Used to encapsulate what would normally be global variables."""
+    def __init__(self, debug, out_dir, no_remove_files):
+        self.debug = debug
+        self.out_dir = out_dir
+        self.tmp_dir = tempfile.mkdtemp()
+        self.pid_file_path = os.path.join(self.out_dir, "profdata_merge_worker.pid")
+        self.final_profdata_path = os.path.join(self.out_dir, "swift.profdata")
+        self.remove_files = not no_remove_files
+
+printlock = Lock()
+def printsync(msg):
+    if CONFIG and CONFIG.debug:
+        with printlock:
+            print >>sys.stderr, msg
```

Comment:
nit-pick: There's been some work done in #90 and #639 to make sure scripts work on both Python 2 and 3. To this end, you should `from __future__ import print_function` at the top of this file, and use the `print()` function here.